### PR TITLE
Fix Agent crash using the Forwarded Events from eventchannel configuration

### LINKED ISSUE #20509: Agent crashes using the `Forwarded Events` from eventchannel configuration
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| ≤ 4.3.0 | wazuh-agent | Agent | any | Windows |

# Description 

There is a problem related to the event channel log format, where the agent collects and forwards events from other Windows machines, this event channel is called `Forwarded Events` and in Wazuh agent configuration it works with `ForwardedEvents`.
The same configuration in the same agent version on Windows 2022 OS fails, the agent service crashes a few seconds after starting, but, if the `ForwardedEvents` section is removed/commented, the agent starts without problems. 
The only difference is the Windows version.

Agent configurations 003 (working) and 008 (not working).
[ossec.conf.agents.zip](https://github.com/wazuh/external-devel-requests/files/12836331/ossec.conf.agents.zip)

Windows versions for each:
```
OS Version Windows Agent003
Microsoft Windows Server 2012 R2 Standard 6.3.9600
PS C:\Users\robert.lee> (Get-WmiObject -Class Win32_OperatingSystem).Version
6.3.9600

OS Version Windows Agent008
Microsoft Windows Server 2022 Datacenter 10.0.20348
PS C:\Users\robert.lee> (Get-WmiObject -Class Win32_OperatingSystem).Version
10.0.20348
```

Wazuh agents and manager version: 4.3.10

### PR DESCRIPTION
**Pull request by nbertoldo**
|Related issue|
|---|
|Closes #20509|

## Description

There is a problem related to the event channel log format, where the agent collects and forwards events from other Windows machines, this event channel is called `Forwarded Events` and in Wazuh agent configuration it works with `ForwardedEvents`.
The same configuration in the same agent version on Windows 2022 OS fails, the agent service crashes a few seconds after starting, but, if the `ForwardedEvents` section is removed/commented, the agent starts without problems. 
The only difference is the Windows version.

Agent configurations 003 (working) and 008 (not working).
[ossec.conf.agents.zip](https://github.com/wazuh/external-devel-requests/files/12836331/ossec.conf.agents.zip)

```xml
<ossec_config>
  <localfile>
    <location>ForwardedEvents</location>
    <log_format>eventchannel</log_format>
    <query>Event/System[EventID != 5145 and EventID != 5156 and EventID != 5447 and
      EventID != 4656 and EventID != 4658 and EventID != 4663 and EventID != 4660 and
      EventID != 4670 and EventID != 4690 and EventID != 4703 and EventID != 4907 and
      EventID != 5152 and EventID != 5157]</query>
  </localfile>
</ossec_config>
```

Windows versions for each:
```
OS Version Windows Agent003
Microsoft Windows Server 2012 R2 Standard 6.3.9600
PS C:\Users\robert.lee> (Get-WmiObject -Class Win32_OperatingSystem).Version
6.3.9600

OS Version Windows Agent008
Microsoft Windows Server 2022 Datacenter 10.0.20348
PS C:\Users\robert.lee> (Get-WmiObject -Class Win32_OperatingSystem).Version
10.0.20348
```

Wazuh agents and manager version: 4.3.10

### CODE REVIEW

File: src/syscheckd/src/whodata/win_whodata.c [hunk 1]
```diff
@@ -1089,7 +1088,8 @@ int policy_check() {
 
 error:
     err_code = GetLastError();
-
+    char err_msg[OS_SIZE_1024];
+    err_msg[OS_SIZE_1024 - 1] = '\0';
```
Comment (nmkoremblum) [inline src/syscheckd/src/whodata/win_whodata.c:1092]: I would recommend changing this in favor of:
```suggestion
    char err_msg[OS_SIZE_1024];
    memset(err_msg, 0, OS_SIZE_1024); // Set all elements to zero
```
I suggest changing it as well in other occurrences.

File: src/logcollector/read_win_event_channel.c [hunk 2]
```diff
@@ -170,7 +182,7 @@ char *get_message(EVT_HANDLE evt, LPCWSTR provider_name, DWORD flags)
         goto cleanup;
     }
 
-    if ((buffer = calloc(size, sizeof(wchar_t))) == NULL) {
+    if ((buffer = calloc(++size, sizeof(wchar_t))) == NULL) {
```
Comment (nmkoremblum) [inline src/logcollector/read_win_event_channel.c:185]: To improve code readability I suggest changing this line as follows:
 ```suggestion
    // Buffer size is increased by one unit as `EvtFormatMessage` does not count the '\0'
    size += 1;
    if ((buffer = calloc(size, sizeof(wchar_t))) == NULL) {
```

**Review by nmkoremblum — CHANGES_REQUESTED** [review]

File: src/unit_tests/logcollector/test_read_win_event_channel.c [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    will_return(wrap_FormatMessage, "File not found.");
+    expect_string(__wrap__mdebug1, formatted_msg, "Could not EvtOpenPublisherMetadata() with flags (1) which returned (2): File not found.");
+
+    assert_null(get_message(evt, provider_name, EvtFormatMessageEvent));
+}
+
+void test_get_message_get_size_fail(void ** state) {
+    EVT_HANDLE evt = NULL;
+    LPCWSTR provider_name = L"provider_name";
+    char message[] = "Test_Message";
```
Comment (LucioDonda) [inline src/unit_tests/logcollector/test_read_win_event_channel.c:62]: Some variables and values of each test are repeated, can this be unified into a common setup stage?

**Review by LucioDonda — CHANGES_REQUESTED** [review]
minor changes regarding UTs

**Review by LucioDonda — DISMISSED** [review]
LGTM!

**Review by nmkoremblum — DISMISSED** [review]
LGTM

**Review by cborla — DISMISSED** [review]
LGTM!

**Pull request closed without merging**